livedata/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt

```diff
@@ -988,42 +997,62 @@ class ChannelControllerImpl(
 
     suspend fun editMessage(message: Message): Result<Message> {
         val online = domainImpl.isOnline()
-        message.updatedAt = Date()
-        message.syncStatus = SyncStatus.IN_PROGRESS
+        val now = Date()
+        var editedMessage = message.copy()
+
+        // set message.updated at if it's null or older than now (prevents issues with incorrect clocks)
+        if (editedMessage.updatedAt == null) {
+            editedMessage.updatedAt = now
+        }
+        editedMessage.updatedAt?.let {
+            if (it.before(now)) {
+                editedMessage.updatedAt = now
+            }
+        }
+
```

Comment:
not happy with this verbosity, what's the shortest way to set `editMessage.updatedAt` to now when it's null or before `now`?

Reply:
added code suggestion above